spec/models/piece_spec.rb

```diff
@@ -55,4 +55,20 @@
       expect(piece.diff(3, piece.x_position)).to eq 3
     end
   end
+
+  describe '.move_to!' do
+    it 'should move the piece to the destination square' do
+      expect(piece.move_to!(2, 6)).to be true
```

Comment:
This test doesn't check that the piece was moved to 2, 6, it's just testing that the move_to! method returns true with those coordinates, which it does when the test is run, because that's technically a `valid_move?` for the superclass. So in this example, it only runs line 56 of the method and returns true, skipping lines 57-58. I think this test needs to be run on an instance of an actual piece type, so that the piece-specific rules are encountered, and then the test would need to check that x_position & y_position now equal the new coordinates.